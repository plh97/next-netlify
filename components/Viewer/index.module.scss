.Viewer {
  width: 100%;

  .container {
    min-width: 550px;
    margin-bottom: 1rem;
    position: relative;
    box-shadow: 0px 0px 5px #0003;
    width: 100%;
    height: 17rem;
    background-image: url(/bg1.jpg);
    background-origin: border-box;
    background-repeat: no-repeat;
    background-size: cover;
    border-radius: 0.5rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    background-position-y: bottom;

    .detail {
      background-color: rgba(245, 245, 245, 0.23);
      flex: 0 0;
      display: flex;
      flex-direction: row;
      padding: 1rem;
      box-sizing: border-box;
      width: 100%;
      align-items: stretch;

      .image {
        width: 4rem;
        height: 4rem;
        border-radius: 1rem;
        margin-right: 1rem;
      }

      .detail-list {
        display: inline-flex;
        flex-direction: column;
        justify-content: space-between;

        .detail-name {
          font-size: 40px;
          line-height: 30px;
        }

        .detail-bio {
          font-size: 16px;
        }
      }
    }
  }
}
