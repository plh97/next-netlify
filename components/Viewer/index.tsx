import React from "react";
import Image from "next/image";
import styles from "./index.module.scss";
import { IViewer } from "@/apis/user";

interface IProps {
  data?: IViewer;
  title: string;
}

export default function Viewer({
  data = {
    avatarUrl: "",
    name: "name",
    bio: "bio",
  },
  title = "",
}: IProps) {
  return (
    <div className={styles["Viewer"]}>
      <div className={styles["container"]}>
        <h1>{title}</h1>
        <div className={styles["detail"]}>
          {data?.avatarUrl && (
            <Image
              className={styles["image"]}
              alt="avatar"
              width={64}
              height={64}
              src={data?.avatarUrl}
            />
          )}
          <span className={styles["detail-list"]}>
            <span className={styles["detail-name"]}>{data.name}</span>
            <span className={styles["detail-bio"]}>{data.bio}</span>
          </span>
        </div>
      </div>
    </div>
  );
}
