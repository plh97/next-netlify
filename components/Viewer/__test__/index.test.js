// Link.react.test.js
import React from 'react'
import renderer from 'react-test-renderer'
import Viewer from '../index.js'

it('renders correctly', () => {
	const props = {
		title: '标题',
		data: {
			avatarUrl: 'http://baidu.com',
			bio: '名言',
			name: '名字'
		}
	}
	const tree = renderer.create(<Viewer {...props} />).toJSON()
	expect(tree).toMatchSnapshot()
})
