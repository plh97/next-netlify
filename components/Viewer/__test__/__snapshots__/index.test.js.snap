// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Viewer"
>
  <div
    className="Viewer__container"
  >
    <h1
      className="Viewer__title"
    >
      标题
    </h1>
    <div
      className="Viewer__detail"
    >
      <img
        alt="avatar"
        className="Viewer__image"
        src="http://baidu.com"
      />
      <span
        className="Viewer__detail-list"
      >
        <span
          className="Viewer__detail-name"
        >
          名字
        </span>
        <span
          className="Viewer__detail-bio"
        >
          名言
        </span>
      </span>
    </div>
  </div>
</div>
`;
